designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/table/ui/TableDimensions.kt

```diff
@@ -0,0 +1,274 @@
+ * @property tableHorizontalPadding The horizontal padding of the table.
+ * @property tableVerticalPadding The vertical padding of the table.
+ * @property defaultCellWidth The default width of the table cells.
+ * @property defaultCellHeight The default height of the table cells.
+ * @property defaultRowHeaderWidth The default width of the row header.
+ * @property defaultHeaderHeight The default height of the header.
+ * @property defaultLegendCornerSize The default size of the legend corner.
+ * @property defaultLegendBorderWidth The default width of the legend border.
+ * @property defaultHeaderTextSize The default size of the header text.
+ * @property defaultRowHeaderTextSize The default size of the row header text.
+ * @property defaultCellTextSize The default size of the cell text.
+ * @property totalWidth The total width of the table.
+ * @property cellVerticalPadding The vertical padding of the table cells.
+ * @property cellHorizontalPadding The horizontal padding of the table cells.
+ * @property headerCellPaddingValues The padding values of the header cells.
+ * @property tableBottomPadding The bottom padding of the table.
+ * @property extraWidths The extra widths of the table.
+ * @property rowHeaderWidths The widths of the row headers.
+ * @property columnWidth The widths of the columns.
+ * @property minRowHeaderWidth The minimum width of the row header.
+ * @property minColumnWidth The minimum width of the column.
+ * @property maxRowHeaderWidth The maximum width of the row header.
+ * @property maxColumnWidth The maximum width of the column.
+ * @property tableEndExtraScroll The extra scroll of the table end.
+ */
+@Immutable
+data class TableDimensions(
+    val tableHorizontalPadding: Dp = Spacing.Spacing16,
+    val tableVerticalPadding: Dp = Spacing.Spacing16,
+    val defaultCellWidth: Int = 160,
+    val defaultCellHeight: Dp = Spacing.Spacing36,
+    val defaultRowHeaderWidth: Int = 275,
+    val defaultHeaderHeight: Int = 36,
+    val defaultLegendCornerSize: Dp = Spacing.Spacing2,
+    val defaultLegendBorderWidth: Dp = Spacing.Spacing8,
+    val defaultHeaderTextSize: TextUnit = 12.sp,
+    val defaultRowHeaderTextSize: TextUnit = 12.sp,
+    val defaultCellTextSize: TextUnit = 12.sp,
+    val totalWidth: Int = 0,
+    val cellVerticalPadding: Dp = Spacing.Spacing4,
+    val cellHorizontalPadding: Dp = Spacing.Spacing4,
+    val headerCellPaddingValues: PaddingValues = PaddingValues(
+        horizontal = Spacing.Spacing4,
+        vertical = Spacing.Spacing11,
+    ),
+    val tableBottomPadding: Dp = Spacing.Spacing200,
+    val extraWidths: Map<String, Int> = emptyMap(),
+    val rowHeaderWidths: Map<String, Int> = emptyMap(),
+    val columnWidth: Map<String, Map<Int, Int>> = emptyMap(),
+    val minRowHeaderWidth: Int = 130,
+    val minColumnWidth: Int = 130,
+    val maxRowHeaderWidth: Int = Int.MAX_VALUE,
+    val maxColumnWidth: Int = Int.MAX_VALUE,
+    val tableEndExtraScroll: Dp = Spacing.Spacing6,
+) {
```

Comment:
Could these be too much properties on the data class, maybe they could be grouped someway. Maybe reducing the amount of properties needed. What do you think?

Reply:
I agree, this issue is just a migration to validate  proper work of the table. Maybe we could improve it in the future